components/search/backend/src/test/kotlin/routes/GetRunsWithPackageIntegrationTest.kt

```diff
@@ -348,6 +353,117 @@ class GetRunsWithPackageIntegrationTest : SearchIntegrationTest({
+
+                response shouldHaveStatus HttpStatusCode.BadRequest
+            }
+        }
+
+        "return BadRequest if neither identifier nor purl is provided" {
+            searchTestApplication { client ->
+                val response = client.get(SEARCH_ROUTE)
+
+                response shouldHaveStatus HttpStatusCode.BadRequest
+            }
+        }
+
+        "return runs globally when searching by purl" {
+            val fixtures = dbExtension.fixtures
+            val packageIdentifier = identifierFor("purl-global")
+            val run = createRunWithPackageForPurlSearch(
+                fixtures = fixtures,
+                repoId = fixtures.repository.id,
+                pkgId = packageIdentifier
+            )
+
+            searchTestApplication { client ->
+                val response = client.get(SEARCH_ROUTE) {
+                    parameter("purl", packageIdentifier.toPurl())
+                }
+
+                response shouldHaveStatus HttpStatusCode.OK
+
+                val body = response.body<List<RunWithPackage>>()
+                body shouldContainExactly listOf(run)
+            }
+        }
+
+        "return curated purl when searching by purl with curation" {
+            val fixtures = dbExtension.fixtures
+            val db = dbExtension.db
+            val packageIdentifier = identifierFor("purl-curated")
+            val curatedPurl = "pkg:maven/corrected-ns/corrected-name@1.0.0"
+            val run = createRunWithCuratedPurl(
+                db = db,
+                fixtures = fixtures,
+                repoId = fixtures.repository.id,
+                pkgId = packageIdentifier,
+                curatedPurl = curatedPurl
+            )
+
+            searchTestApplication { client ->
+                val response = client.get(SEARCH_ROUTE) {
+                    parameter("purl", curatedPurl)
+                }
+
+                response shouldHaveStatus HttpStatusCode.OK
+
+                val body = response.body<List<RunWithPackage>>()
+                body shouldContainExactly listOf(run)
+            }
+        }
+
+        "return packageId as null and purl as non-null for purl search" {
+            val fixtures = dbExtension.fixtures
+            val packageIdentifier = identifierFor("purl-response")
+            createRunWithPackageForPurlSearch(
+                fixtures = fixtures,
+                repoId = fixtures.repository.id,
+                pkgId = packageIdentifier
+            )
+
+            searchTestApplication { client ->
+                val response = client.get(SEARCH_ROUTE) {
+                    parameter("purl", packageIdentifier.toPurl())
+                }
+
+                response shouldHaveStatus HttpStatusCode.OK
+
+                val body = response.body<List<RunWithPackage>>()
+                body.size shouldBe 1
+                body.first().packageId.shouldBeNull()
+                body.first().purl shouldBe packageIdentifier.toPurl()
```

Comment:
This can be simplified to

                body.shouldBeSingleton {
                    it.packageId.shouldBeNull()
                    it.purl shouldBe packageIdentifier.toPurl()
                }

Similar for 3 other places that use the `size shouldBe 1` approahc.